srcts/src/components/webcomponents/popover.ts

```diff
@@ -385,7 +385,7 @@ export class BslibPopover extends BslibElement {
       trigger: this.triggerElement,
       content: {
         // eslint-disable-next-line @typescript-eslint/naming-convention
-        ".popover-header": hasHeader(newHeader) ? newHeader : "",
+        ".popover-header": header && header.html ? newHeader : "",
```

Comment:
This is one of those cases where I think an intermediate variable would be helpful just from a readability perspective.

e.g. something like this

```suggestion
        ".popover-header": has_updated_header ? newHeader : "",
```

where

```js
has_updated_header: header && header.html
```

comes very early on, maybe even just after `header` is destructured from the message data.

That said, I'm confused by this comment above with `getOrCreateElement()`. 

https://github.com/rstudio/bslib/blob/3941bc7ee7985e8fc12a3d39d664900758bdf045/srcts/src/components/webcomponents/popover.ts#L357-L361

I guess I'm confused why `header` goes through `getOrCreateElement()` if we end up ignoring its value when `header.html` exists?

Reply:
One other note, I'm guessing this behavior was introduced by changes in #747, but I'm having a hard time seeing how the bug was introduced or how this fixes it.

Normally, for a one-line change for a bug like this, I'd expect the fix to feel a little more obvious. But this feels _subtle_. I'm not sure you need to do anything with that information, other than maybe to explain why this is the fix in this PR.

Reply:
Thanks for the great observations and apologies for throwing up this half-baked PR late on a Friday evening (wanted to jot my thoughts down, but should've left as a draft). 

After coming back to this, I'm embarrassed to say the proposed fix here is way off-basis: `update_popover(header = NULL)` shouldn't remove the header, only `update_popover(header = character(0))` should. The testing app that surfaced this issue should be passing `character(0)` but is currently passing `NULL`, and it was only ever working before due to a weird interaction with #747.

Reply:
No worries, thanks for the update!